.github/workflows/test_suite_ubuntu_cuda.yml

```
@@ -58,15 +62,15 @@ jobs:
       - name: Install Python
         uses: actions/setup-python@v5
         with:
-          python-version: '3.x'
+          python-version: '3.13'  # Use 3.13 as no PyTorch wheels for 3.14 yet.
 
       - name: Install PyTorch
         run: |
           python -m pip install --upgrade pip
           python -m venv ftorch
           . ftorch/bin/activate
-          # CUDA v11.8 chosen as the GPU-runner has v12.4 installed (backward compatible)
-          pip install torch torchvision --index-url https://download.pytorch.org/whl/cu118
+          # CUDA v12.9 chosen as latest - check what the GPU-runner has installed (backward compatible)
```

Does this mean, we have select v12.9 because the GPU instance has installed 12.9 drivers? If so could we re-word it to something like:

```suggestion
          # CUDA v12.9 chosen to match version installed by GPU-runner
```

Or did we just pick it because (as you mention) it's backwards compatible so it doesnt really matter. In which case I dont think we need the comment at all :shrug:

---

The original comment from @niccolozanotti suggested the runners may not have latest CUDA.
I decided to start with 12.9 to keep us as far ahead of the curve, planning to iterate backwards, but it seems 12.9 works 🤷